commands/volumes/upload.go

```diff
@@ -56,28 +58,64 @@ func handleUploadVolumeCmd(c client.Client) error {
 		params["Org"] = c.Org
 	}
 
+	var file *os.File
+	var err error
+
+	filename := viper.GetString("path")
+	fi, err := os.Stat(filename)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case fi.IsDir():
+		err = zip.CreateArchiveFromDir(filename)
+		if err != nil {
+			return err
+		}
+		file, err = os.Open(filename + ".tar.bz2")
+		if err != nil {
+			return err
+		}
+	case isRegularFile(filename):
+		err = zip.CreateArchiveFromFile(filename)
+		if err != nil {
+			return err
+		}
+		file, err = os.Open(filename + ".tar.bz2")
+		if err != nil {
+			return err
+		}
+	default: // .bz2 file
+		file, err = os.Open(filename)
+		if err != nil {
+			return err
+		}
+	}
+
+	defer os.Remove(filename + ".tar.bz2")
+
 	subresource := fmt.Sprintf("volume/%s/upload", volume)
 	url := uri.CreateResourceURI("", "environment", envID, subresource, params)
-	form, contentType, err := fileForm(viper.GetString("file"), "volume_tarball")
+	form, contentType, err := fileForm(file, "volume_tarball")
 	if err != nil {
 		return err
 	}
 	_, err = c.Requester.Do(http.MethodPost, url, contentType, form)
 	return err
 }
 
-func fileForm(path, formField string) (*bytes.Buffer, string, error) {
-	file, err := os.Open(path)
-	if err != nil {
-		return nil, "", err
-	}
-	defer file.Close()
+func isRegularFile(path string) bool {
+	ext := filepath.Ext(path)
+	return ext != ".bz2"
```

Comment:
This isn't accurate or is misnamed. Did you mean something like this?

```
fileInfo, err := os.Stat(path)
if err != nil {
  something
}

return fileInfo.IsDir() 
```

Reply:
Renamed. This checks if a file at a given path is a .bz2 file.